File: 2-year/OOP-CSharp/src/Lab5/Presentation/Lab5.Presentation.Console/Scenarios/GetHisotry/GetHistoryScenario.cs
using Lab5.Application.Contracts.Users;
using Lab5.Application.Models.Users;
using Spectre.Console;
using System.Globalization;

namespace Lab5.Presentation.Console.Scenarios.GetHisotry;

public class GetHistoryScenario : IScenario
{
    private readonly IUserService _userService;

    public GetHistoryScenario(IUserService userService)
    {
        _userService = userService;
    }

    public string Name => "Get history";

    public void Run()
    {
        IEnumerable<UserOperation> result = _userService.GetOperationsHistory();
        var table = new Table();
        table.AddColumn("OperationId");
        table.AddColumn("AccountId");
        table.AddColumn("MoneyAmount");
        table.AddColumn("OperationName");

        foreach (UserOperation operation in result)
        {
            table.AddRow(
                operation.OperationId.ToString(),
                operation.AccountId.ToString(),
                operation.MoneyAmount.ToString(CultureInfo.CurrentCulture),
                operation.OperationName);
        }

        AnsiConsole.Write(table);
        AnsiConsole.Ask<string>("Ok");
    }
}

File: 2-year/Java/Lab1/src/main/java/com/Starygin/Presentation/Scenarios/GetHistoryScenario.java
package com.Starygin.Presentation.Scenarios;

import com.Starygin.Application.Contracts.UserService;
import com.Starygin.Application.Models.UserOperation;
import com.Starygin.Presentation.Scenario;

import java.util.List;

public class GetHistoryScenario implements Scenario {
    private final UserService userService;

    public GetHistoryScenario(UserService userService) {
        this.userService = userService;
    }

    @Override
    public String getName() {
        return "Get History";
    }

    @Override
    public void run() {
        List<UserOperation> history = userService.getOperationsHistory();
        System.out.println(history);
    }
}
